consensus/src/alerts/service.rs

```diff
@@ -5,69 +5,73 @@ use crate::{
 
 pub struct Service<H: Hasher, D: Data, MK: MultiKeychain> {
     messages_for_network: Sender<(NetworkMessage<H, D, MK>, Recipient)>,
     messages_from_network: Receiver<NetworkMessage<H, D, MK>>,
     notifications_for_units: Sender<ForkingNotification<H, D, MK::Signature>>,
     alerts_from_units: Receiver<Alert<H, D, MK::Signature>>,
-    rmc: ReliableMulticast<H::Hash, MK>,
-    messages_for_rmc: Sender<RmcMessage<H::Hash, MK::Signature, MK::PartialMultisignature>>,
-    messages_from_rmc: Receiver<RmcMessage<H::Hash, MK::Signature, MK::PartialMultisignature>>,
-
     data_for_backup: Sender<AlertData<H, D, MK>>,
     responses_from_backup: Receiver<AlertData<H, D, MK>>,
-
     own_alert_responses: HashMap<H::Hash, OnOwnAlertResponse<H, D, MK>>,
     network_alert_responses: HashMap<H::Hash, OnNetworkAlertResponse<H, D, MK>>,
     multisigned_notifications: HashMap<H::Hash, ForkingNotification<H, D, MK::Signature>>,
-
     node_index: NodeIndex,
     exiting: bool,
+    handler: Handler<H, D, MK>,
+    rmc_service: RmcService<H::Hash, MK, MK::Signature, MK::PartialMultisignature>,
+}
+
+pub struct IO<H: Hasher, D: Data, MK: MultiKeychain> {
+    pub messages_for_network: Sender<(NetworkMessage<H, D, MK>, Recipient)>,
+    pub messages_from_network: Receiver<NetworkMessage<H, D, MK>>,
+    pub notifications_for_units: Sender<ForkingNotification<H, D, MK::Signature>>,
+    pub alerts_from_units: Receiver<Alert<H, D, MK::Signature>>,
+    pub data_for_backup: Sender<AlertData<H, D, MK>>,
+    pub responses_from_backup: Receiver<AlertData<H, D, MK>>,
 }
 
 impl<H: Hasher, D: Data, MK: MultiKeychain> Service<H, D, MK> {
-    pub fn new(
-        keychain: MK,
-        messages_for_network: Sender<(NetworkMessage<H, D, MK>, Recipient)>,
-        messages_from_network: Receiver<NetworkMessage<H, D, MK>>,
-        notifications_for_units: Sender<ForkingNotification<H, D, MK::Signature>>,
-        alerts_from_units: Receiver<Alert<H, D, MK::Signature>>,
-        data_for_backup: Sender<AlertData<H, D, MK>>,
-        responses_from_backup: Receiver<AlertData<H, D, MK>>,
-    ) -> Service<H, D, MK> {
-        let (messages_for_rmc, messages_from_us) = mpsc::unbounded();
-        let (messages_for_us, messages_from_rmc) = mpsc::unbounded();
+    pub fn new(keychain: MK, io: IO<H, D, MK>, handler: Handler<H, D, MK>) -> Service<H, D, MK> {
+        let IO {
+            messages_for_network,
+            messages_from_network,
+            notifications_for_units,
+            alerts_from_units,
+            data_for_backup,
+            responses_from_backup,
+        } = io;
 
-        let rmc = ReliableMulticast::new(
-            messages_from_us,
-            messages_for_us,
-            keychain.clone(),
-            DoublingDelayScheduler::new(time::Duration::from_millis(500)),
+        let node_index = keychain.index();
+        let rmc_handler = aleph_bft_rmc::Handler::new(keychain);
```

Comment:
Considering what we talked about earlier it's unlikely you will end up creating this handler here, but I guess you can extract it into its proper place in the next PR. Feel free to resolve if this is your plan.

Reply:
yep, constructing handlers in proper places remains for the next pr